### PR TITLE
Use only `city` and `state` on the front end

### PR DESCRIPTION
**Pull request by cvalarida**
This should simplify handling the city and state fields.

The idea is that we'll translate their military address equivalent when talking to the api, but keep them as `city` and `state` in the front-end code.

### CODE REVIEW

**Comment by cvalarida** [comment]
I ended up skipping the tests 😞 

We can circle back around to this later in [this ticket](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/4961).

File: src/js/letters/utils/constants.js [hunk 1]
```diff
@@ -133,3 +133,9 @@ export const STATE_CODE_TO_NAME = {
   PI: 'Virgin Islands'
 };
 
+export const addressTypes = Object.freeze({
```
Comment (annekainicUSDS) [inline src/js/letters/utils/constants.js]: Is it weird to add a constant variable that's not capitalized? I don't know too much about the conventions here, but that seems to be what we're following for the other constants.

File: src/js/letters/actions/letters.js [hunk 2]
```diff
@@ -215,10 +227,18 @@ export function saveAddressFailure(address) {
 }
 
 export function saveAddress(address) {
+  const transformedAddress = Object.assign({}, address);
+  if (transformedAddress.type === addressTypes.military) {
```
Comment (annekainicUSDS) [inline src/js/letters/actions/letters.js]: I assume the code to actually set the `type` before saving will be somewhere else?

File: src/js/common/helpers/api.js [hunk 3]
```diff
@@ -37,5 +37,6 @@ export function apiRequest(resource, optionalSettings = {}, success, error) {
 
       return data;
     })
-    .then(success, error);
+    .then(success)
+    .catch(error);
```
Comment (annekainicUSDS) [inline src/js/common/helpers/api.js]: Is there a reason for this change? It behaves the same I assume, but might just be more descriptive of what's actually happening?

**Review by annekainicUSDS — COMMENTED** [review]
LGTM! Just a few questions

File: src/js/letters/utils/constants.js (hunk 1 above)
Comment (cvalarida) [reply src/js/letters/utils/constants.js]: Yeah, that's fair. I can capitalize it. 👍

File: src/js/letters/actions/letters.js (hunk 2 above)
Comment (cvalarida) [reply src/js/letters/actions/letters.js]: Yup! In another PR.

File: src/js/common/helpers/api.js (hunk 3 above)
Comment (cvalarida) [reply src/js/common/helpers/api.js]: Yeah, this one was tricky. Basically, if the `success` function threw an error for whatever reason, it wouldn't be caught by `error`. I was running into that while trying to mock out the `fetch`.

Comment (annekainicUSDS) [reply src/js/common/helpers/api.js]: Ah okay, good catch!

**Review by annekainicUSDS — APPROVED** [review]
LGTM!

**Pull request merged**